tests/FakeItEasy.Specs/ManageEventsSpecs.cs

```diff
@@ -0,0 +1,140 @@
+        }
+
+        [Scenario]
+        public void ManageSpecificEventUnsubscribe(IFoo fake, EventHandler handler)
+        {
+            "Given a strict fake"
+                .x(() => fake = A.Fake<IFoo>(o => o.Strict()));
+
+            "And an event handler"
+                .x(() => handler = A.Fake<EventHandler>());
+
+            "And the fake is configured to manage the Bar event"
+                .x(() => Manage.Event(nameof(IFoo.Bar)).Of(fake));
+
+            "When the handler is subscribed to the Bar event"
+                .x(() => fake.Bar += handler);
+
+            "And the handler is unsubscribed from the Bar event"
+                .x(() => fake.Bar -= handler);
+
+            "And the Bar event is raised using Raise"
+                .x(() => fake.Bar += Raise.WithEmpty());
+
+            "Then the handler is not called"
+                .x(() => A.CallTo(handler).MustNotHaveHappened());
+        }
+
+        [Scenario]
+        public void ManageSpecificEventDoesntHandlerOtherEvents(IFoo fake, EventHandler handler, Exception exception)
+        {
+            "Given a strict fake"
+                .x(() => fake = A.Fake<IFoo>(o => o.Strict()));
+
+            "And an event handler"
+                .x(() => handler = A.Fake<EventHandler>());
+
+            "And the fake is configured to manage the Bar event"
+                .x(() => Manage.Event(nameof(IFoo.Bar)).Of(fake));
+
+            "When the handler is subscribed to the Baz event"
+                .x(() => exception = Record.Exception(() => fake.Baz += handler));
+
+            "Then it throws an ExpectationException"
+                .x(() => exception.Should().BeAnExceptionOfType<ExpectationException>());
+        }
+
+        [Scenario]
+        public void ManageAllEvents(IFoo fake, EventHandler barHandler, EventHandler bazHandler)
+        {
+            "Given a strict fake"
+                .x(() => fake = A.Fake<IFoo>(o => o.Strict()));
+
+            "And event handlers for Bar and Baz"
+                .x(() => (barHandler, bazHandler) = (A.Fake<EventHandler>(), A.Fake<EventHandler>()));
+
+            "And the fake is configured to manage any event"
+                .x(() => Manage.AllEvents.Of(fake));
+
+            "When the handler for Bar is subscribed to the Bar event"
+                .x(() => fake.Bar += barHandler);
+
+            "And the handler for Baz is subscribed to the Baz event"
+                .x(() => fake.Baz += bazHandler);
+
+            "And the Bar event is raised using Raise"
+                .x(() => fake.Bar += Raise.WithEmpty());
+
+            "And the Baz event is raised using Raise"
+                .x(() => fake.Baz += Raise.WithEmpty());
+
+            "Then the handler for Bar is called"
+                .x(() => A.CallTo(barHandler).MustHaveHappened());
+
+            "And the handler for Baz is called"
+                .x(() => A.CallTo(bazHandler).MustHaveHappened());

```

Comment:
I believe that each raised event went to the right handler, but we could augment the test to ensure it's so.

Reply:
Ah, yes. I'll look into it. Passing specific event args should be enough to ensure this.